### DISCUSSION TITLE
TypeError: Deno. open is not a function

### DISCUSSION THREAD

**Discussion by Kikkon** [supabase/supabase#20415] [Questions] [answered]
I am trying to run WASM on my edge function using Extism (https://extism.org/). If I try to load a local WASM file directly, I get the error "Deno.open is not a function". I think this is because edge functions are not allowed to access local files, so there is a runtime restriction?

**Comment by Kikkon** [comment]
my test code 
```js
import createPlugin from 'https://raw.githubusercontent.com/extism/js-sdk/main/src/mod.ts';
Deno.serve(async (req) => {
  const { name } = await req.json()
  const buffer = await fetch("https://esm.sh/@dqbd/tiktoken@1.0.3/lite/tiktoken_bg.wasm");
  const plugin = await createPlugin(
    buffer.arrayBuffer(),{ useWasi: true }
  );
  let out = await plugin.call("count_vowels", name);
  return new Response(
    JSON.stringify(out),
    { headers: { "Content-Type": "application/json" } },
  )
})
```

 I am not familiar with Deno and TypeScript.

**Reply by GaryAustin1** [reply, marked as answer]
![image](https://github.com/supabase/supabase/assets/54564956/d97cf939-7e71-4c88-94f8-4eb8d37b81eb)
https://supabase.com/docs/guides/functions/debugging
